packages/runtime/src/unknown-types.ts

```diff
@@ -44,10 +44,12 @@ export type UnknownEnum = number;
  * A unknown oneof group. See `isOneofGroup()` for details.
  */
 export type UnknownOneofGroup = {
-    oneofKind: undefined | string;
-    [k: string]:
+    kind: undefined;
+    value?: never;
+} | {
+    kind: string;
+    value:
         | UnknownScalar
         | UnknownEnum
-        | UnknownMessage
-        | undefined;
+        | UnknownMessage;
```

Comment:
Do you think removing `undefined` here is the right call? 

Users should be able to access value like this:

```ts
const v: string | number | undefined = example.result.value;
```

So having `undefined` as part of the union seems to make sense.

Reply:
I thought that if the `kind` was set that the `value` also had to be set. The big advantage is when trying to switch on the `kind` you can access the non-nullable `value` for any string `case`. Compared to before where even if you were inside a string `case` you still were getting a potentially undefined `value` even though (as far as I know) the value could never be undefined after that refinement.

To be clear, your example should still work just fine (see [playground](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true#code/C4TwDgpgBAqgdgazgewO5wPJwsgZgcQCdkBXMKAXigG8AoKBqBASzgBMAuKE9iXViGwDc9RgDcAhgBsSEAPxdsYiIREBfKAB8aohi3ZcAzsEKsA5iMZRJMiEZPn1I2mwgBjKRMLQ3yOMahcZGQueCQ0TGw8IlIwZ19-YCgJe1M4My1uXn5sNkpA4IA6G1khIA)).

Reply:
Ah, completely missed that `UnknownOneofGroup` is a union now instead of an indexed type. That makes sense.